src/platformAccessory.ts

```diff
@@ -25,16 +31,19 @@ export class Car {
   private engine: Service;
   private lock: Service;
   private target: Target;
+  private config: CarConfig;
 
   constructor(
     private readonly platform: Platform,
     private readonly accessory: PlatformAccessory,
     private readonly kiaConnect: KiaConnect,
     private readonly name: string,
     private readonly vin: string,
-    private readonly targetTemperature: string,
-    private readonly refreshInterval: number,
+    config: Partial<CarConfig>,
   ) {
+    // Initialize config to default values
+    this.config = initializeConfig(config);
```

Comment:
We will take the input from the user specified config, and make sure every option is either set to the user provided value, or some default.